client/app/scripts/selectors/chartSelectors.js

```diff
@@ -0,0 +1,124 @@
+import debug from 'debug';
+import { createSelector, createSelectorCreator, defaultMemoize } from 'reselect';
+import { Map as makeMap, is, Set } from 'immutable';
+
+import { getAdjacentNodes } from '../utils/topology-utils';
+
+
+const log = debug('scope:selectors');
+
+
+//
+// "immutable" createSelector
+//
+const createDeepEqualSelector = createSelectorCreator(
+  defaultMemoize,
+  is
+);
+
+
+const identity = v => v;
+
+
+const allNodesSelector = state => state.get('nodes');
+
+
+export const nodesSelector = createSelector(
```

Comment:
`createSelector` checks its arguments (state.get('nodes') in this case) not the result. we could wrap it in the createDeepEqualSelector+identity dance to check its result however! good point.